android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFcmMessagingService.java

```diff
@@ -31,7 +38,40 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
       return;
     }
 
+    ExponentDB.experienceIdToExperience(remoteMessage.getData().get("experienceId"), new ExponentDB.ExperienceResultListener() {
+      @Override
+      public void onSuccess(ExperienceDBObject experience) {
+        try {
+          JSONObject manifest = new JSONObject(experience.manifest);
+          JSONObject androidSection = manifest.optJSONObject("android");
+          if (androidSection != null) {
+            boolean useNewNotificationsAPI = androidSection.optBoolean("useNewNotificationsAPI", false);
+            if (useNewNotificationsAPI) {
```

Comment:
Another conditional here would be sth like `|| (experience.sdkVersion >= 39 && < UNVERSIONED)`, right? Do you think we should add it already? 🤔